modules/nf-core/sourmash/compare/main.nf

```diff
@@ -0,0 +1,43 @@
+process SOURMASH_COMPARE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.5.0--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.5.0--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(signatures)
+    val save_numpy_matrix
+    val save_csv
+
+    output:
+    tuple val(meta), path("*.comp")            , optional:true, emit: numpy_matrix
+    tuple val(meta), path("*.comp.labels.txt") , optional:true, emit: labels
+    tuple val(meta), path("*.comp.csv")        , optional:true, emit: csv
+    path "versions.yml"                        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args     ?: ''
+    def prefix = task.ext.prefix   ?: "${meta.id}"
+    def comp   = save_numpy_matrix ?: "--output ${prefix}.comp"  : '' 
```

Comment:
For numpy arrays, the [typical file extension](https://numpy.org/doc/stable/reference/generated/numpy.save.html) is `.npy`. Don't know if this is of type array or matrix but either way I think a file extension will be useful.

Reply:
Ooh I hadn't seen that before, I agree this would be useful sign posting and would better tie together the labels.txt file -- if I do `${prefix}.npy` for the numpy matrix, the labels files will automatically become `${prefix}.npy.labels.txt`. I'll update these names once the tests are passing!